### ISSUE TITLE
Setting location is not accessible

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#5991]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

The current method for setting the location via the UI requires visual interaction with a pointer to set latitude and longitude.  This is not accessible for screen reader users  or other assistive technology requiring keyboard control.

## Expected behavior

There needs to be a keyboard only method for setting location either via coordinates or physical address.

## Steps to reproduce

N/A

## Environment

- Home Assistant release with the issue: Issue started with intro of map method for location and persists in latest release.
- Browser and browser version: N/A
- Operating system: N/A

**Comment by steverep** [comment]
Yes, this is certainly still an issue.

**Comment by steverep** [comment]
Yes, still an issue.  And could probably be easily solved simply by adding input fields to show/edit the longitude and latitude coordinates.

**Comment by steverep** [comment]
Yep, still an issue.

**Comment by steverep** [comment]
Please re-open.  This is still an issue.

**Comment by steverep** [comment]
Yes, still an issue.

**Comment by spacegaier** [comment]
@matthiasdebaat Any ideas how to handle this?

**Comment by matthiasdebaat** [comment]
We could use the same pattern as we have in the onboarding? Where there's a button to detect your location.

_Sorry for the dutch language_
![screenshot](https://user-images.githubusercontent.com/20203284/133974750-2b4fb70d-37e3-48a6-8ca3-451d3b85c9a7.png)

**Comment by spacegaier** [comment]
I just thought about it some more: What if the home location cannot reliably be detected automatically? Not sure how often that happens though.

For manually created zones, we have this UI where you have input fields:
![image](https://user-images.githubusercontent.com/114137/133976478-b74fd312-de21-44c6-84fa-9b80c36c865a.png)

**Comment by matthiasdebaat** [comment]
It's good to offer this option, this helps a user to specify his exact location. I think it's a bit hard to use this with screen reader or other assistive technology. @steverep do you have experience with this?

**Comment by steverep** [comment]
Can we stop the bot from closing this?

**Comment by frenck** [comment]
By responding before it closes :)

**Comment by steverep** [comment]
It's Home Assistant not Home Ruler. Playing by the bot's rules makes for a boring movie plot.

**Comment by steverep** [comment]
Please re-open. This is certainly still an issue.

**Comment by steverep** [comment]
Yup, still a problem.

**Comment by Nardol** [comment]
It is still an issue

**Comment by joshms123** [comment]
This looks like it has been an issue for a while. Is anybody looking into it?

**Comment by steverep** [comment]
@joshms123 there are quite a few accessibility issues, some hanging around for a while like this one.  I've been trying to improve accessibility one issue at a time through various contributions, but I'm not part of the HA team so my time is limited.  At least this one has a workaround to use Yaml.

Specific to this issue,inputting coordinates is a start and is possible in the latest version, but I wouldn't regard it as equal access until I can search for an address via keyboard.

**Comment by steverep** [comment]
@bramkragten @zsarnett can fixing this finally be made a priority?  The ability to add/edit zone locations by inputting latitude and longitude now exists for other zones, but cannot be done for the home zone.  That seems pretty silly.

Since the focus of this year is supposed to be streamlining experiences, it would be nice to simply be able to quickly enter physical addresses rather than fumbling through a map.  This is one of the first things new HA users need to do, so it should be as easy and accessible as possible.

Am I missing a technical hurdle?

**Comment by johanhallberg** [comment]
I agree, this "should" be fixed as new HA users (and old ones like me) really like to control this - preferably without yaml-code.  The radius parameter is really something very useful, and especialy for your primary location as home... 😁

I hope this is not too hard as all other zones got this?

**Comment by Shadowex3** [comment]
@johanhallberg agreed. It's astounding to me that this isn't an option, and even more incomprehensible that it _was_ in the past and then somehow the decision was made to deliberately cripple and remove functionality for apparently no reason at all. It's the sort of thing I'd expect from Apple perhaps before they sell you a new device with the crippled feature re-added at extra cost. 

The default size of the "home" zone is positively enormous too. Unless you live in the midwest on some acreage that's going to be several houses or an entire city block.

**Comment by rekinet** [comment]
It doesn't take much to solve this problem. All that's needed is the ability to enter the latitude and longitude. This can be done for other locations, but there has never been such an option for home location. The problem is very old. Maybe it's time to finally solve it?

**Comment by KTibow** [comment]
Wasn't there a new location picker that made this accessible?

**Comment by steverep** [comment]
> Wasn't there a new location picker that made this accessible?

Only for on boarding, and honestly, I haven't even tested that yet

**Comment by lindhe** [comment]
Calling this an accessibility issue is really belittling the issue. I have no disabilities and I struggled for 10+ minutes trying to figure out how to change the location on the map. 😂 Drag-and-dropping great distances is pretty usual UX, I think.

**Comment by steverep** [comment]
> Calling this an accessibility issue is really belittling the issue. 

Are you sure you want to use the word "belittling" here?  In doing so, you are implying that accessibility issues should be treated with less importance than others, which certainly some users would find pretty offensive.



> I have no disabilities and I struggled for 10+ minutes trying to figure out how to change the location on the map. 😂 Drag-and-dropping great distances is pretty usual UX, I think.

Improving drag and drop has nothing to do with this issue.

**Comment by D3SL** [comment]
@steverep You know those memes about someone saying they like coca cola and then getting dogpiled by a bunch of people accusing them of hating pepsi? Anyone acting in good faith can very clearly see @lindhe was using humor to express just how significant of an impediment to any meaningful use of the system this is.

At no point would any reasonable person acting in good faith think he was demeaning accessibility issues.

**Comment by steverep** [comment]
@D3SL please don't accuse me of being unreasonable or acting in bad faith, or lecture me on what should or should not be considered offensive.  I'm perfectly capable of handling that on my own.  My comment certainly gave every benefit of the doubt deserved, and simply offered a perspective and chance to reconsider the choice of wording for whatever reason.

I didn't find "belittling" humorous, and I know I wouldn't be alone in that.  If you did, then you either need a dictionary or perhaps an introspective moment to step back and consider why.

Lastly, these comments have nothing to do with the issue, so I will hide them and request moving on before it diverges further.

**Comment by karwosts** [comment]
Made some updates in #19994 which may address some of the points here. I don't think if it has everything you asked for (no addresses), but it should be more accessible.

**Comment by lindhe** [comment]
Ping

**Comment by lindhe** [comment]
Ping

**Comment by steverep** [comment]
> @steverep - what are you still looking for here? For some time locations (including home zone) have been settable by entering latitude/longitude via the keyboard. 

That was a step forward over having to use YAML, but it still requires looking up the coordinates outside of HA first.  The ability to find the desired coordinates on the map is still only available two sighted mouse users.

Being able to search for a location via address or landmark as in onboarding would close this. implementing them has been on my list for quite some time, but my time to contribute has gone way down lately.

**Comment by bramkragten** [comment]
The problem here is that we got permission from open street map to use their API to do geocoding,  but we currently only have permission to use it during onboarding to limit the number of requests we are sending. We would first need permission from open street map before we could implement this in every location selector.

**Comment by steverep** [comment]
Having it only available during onboarding is really borderline useful for HA.  That amounts to more or less a single lifetime request per user, and there are probably thousands of users like myself who have never used it in production because it was implemented after they were already up and running.

Also, consider that while this issue deals purely with setting zone locations, in reality **all** Geo-tracking info is inaccessible at the moment because it is only conveyed visually on a map.  Reverse geocoding is the only way to make it accessible.

Lastly, note there exists a subset of HA users who, given the option, would happily turn off any external API requests over privacy concerns.

**Comment by tdwalton** [comment]
> [@steverep](https://github.com/steverep) - what are you still looking for here? For some time locations (including home zone) have been settable by entering latitude/longitude via the keyboard.

I'm not sure if my issue belongs here, or if I should file a new bug, but thought I would mention it here since this issue is still open.

I was recently attempting to edit the home zone by inputting coords on an iOS device. The keyboard that pops up on an iOS device, for those input boxes, is the default number keyboard that only has 0 - 9 and a decimal point. Obviously this won't work to enter coordinates that are preceded by a minus sign, so I was forced to paste the coordinates in after typing them out in a note. 

Obviously an edge-case that few will likely encounter, but it does seem also related to accessibility, and as long as there is work in progress for this issue, maybe that could also be addressed here.

**Comment by steverep** [comment]
@tdwalton that should be a separate issue
